tools/integration_tests/concurrent_operations/high_cpu_concurrent_listing_test.go

```diff
@@ -0,0 +1,522 @@
+// Copyright 2024 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package concurrent_operations
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"runtime"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/operations"
+	"github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/setup"
+	"github.com/googlecloudplatform/gcsfuse/v2/tools/integration_tests/util/test_setup"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	requiredCPUCoresToRunThisTest = 30
+)
+
+////////////////////////////////////////////////////////////////////////
+// Boilerplate
+////////////////////////////////////////////////////////////////////////
+
+// This test-suite contains parallelizable test-case. Use "-parallel n" to limit
+// the degree of parallelism. By default it uses GOMAXPROCS.
+// Ref: https://stackoverflow.com/questions/24375966/does-go-test-run-unit-tests-concurrently
+type highCpuConcurrentListingTest struct{}
+
+func (s *highCpuConcurrentListingTest) Setup(t *testing.T) {
+	testDirPath = setup.SetupTestDirectory(testDirName)
+}
+
+func (s *highCpuConcurrentListingTest) Teardown(t *testing.T) {}
+
+func createDirectoryStructureForTestCaseParallel(t *testing.T, testCaseDir string) {
```

Comment:
Can you add a comment explaining the directory structure that's created by this method?